### PR TITLE
add example of using Pulsar with testcontainers-go

### PR DESCRIPTION
**Pull request by clive-jevons**
A quick example of how I used Pulsar in a recent test for another project which others might find useful.

It essentially mimics the way the testcontainers-java Pulsar module sets up Pulsar.

### CODE REVIEW

**Comment by mdelapenya** [comment]
This is super cool!! Thanks for adding the example.

I'd remove the gomega dependency to avoid associating the example with a test framework. I'd use testing library + require or assert. You can check the ones used in the project as a reference.

In any case, this LGTM!!

**Comment by mdelapenya** [comment]
I've just realized that the example is not shown in the website because the mkdocs descriptor living in the root for of the project, she's not included the new entry. Could you please add it?

File: docs/examples/pulsar.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"io"
+	"testing"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func TestPulsar(t *testing.T) {
```
Comment (mdelapenya) [inline docs/examples/pulsar.md:18]: I think it's fine to have the log consumer in place to demonstrate how to do a test.

What other examples do is to create a method to setup the container (setupPulsar) which only creates the container, and then they create a TestFoo function with the real test.

I think that would be even more helpful, as we decouple the pulsar container creation from its usage in a test. Wdyt?

Comment (clive-jevons) [reply docs/examples/pulsar.md]: Done :)

**Review by mdelapenya — APPROVED** [review]
LGTM!!

I've test it locally with success, thanks for this great example on Apache Pulsar usage. Great job

**Pull request merged**